recipes/python/retail/retail/trainingdataloader.py

```diff
@@ -44,6 +46,10 @@ def load(configProperties):
         dataframe = prodreader.load(data_set_id=training_data_set_id,
                              ims_org=configProperties['ML_FRAMEWORK_IMS_TENANT_ID'])
 
+    if '_id' in dataframe.columns:
```

Comment:
We should agree on 'df' or 'dataframe' and better yet, identical code should be in a util method or common class.

Reply:
@rjrussell77 
Yes, I have renamed it to dataframe.